✨Elliptic Curve Cryptography

raxhvl opened the pull request:
Foundations of ECC. Preview the markdown to render media.

taxmeifyoucan reviewed (commented):
Thank you so much for the contribution @raxhvl! Incredible job on this one, it's comprehensive and understandable. I would maybe do some tweaks to the intro but it's great writing! Keep them coming :)

raxhvl commented:
> Thank you so much for the contribution @raxhvl! Incredible job on this one, it's comprehensive and understandable. I would maybe do some tweaks to the intro but it's great writing! Keep them coming :)

Thank your for the kind words! Now the real challenge is writing an intro that lives up to the rest (wink wink).

raxhvl commented on docs/wiki/Cryptography/elliptic-curve-cryptography.md:
````diff
@@ -0,0 +1,283 @@
+-7930066429007744594
+```
+
+2. For every signature, a random **ephemeral key pair [$eK$, $eP$]** is generated to mitigate an [attack](https://youtu.be/DUGGJpn2_zY?si=4FZ3ZlQZTG9-eah9&t=2117) exposing her **secret key 🔑**.
````
Intersting.. It works fine without the space for me.

![Screenshot from 2024-02-27 12-04-21](https://github.com/eth-protocol-fellows/protocol-studies/assets/10168946/f85396f3-d9ca-4bd2-8327-43bccf962f85)

@taxmeifyoucan could you check locally?

SiddharthV1 replied:
I checked locally , it renders fine without the space

raxhvl replied:
Thanks @SiddharthV1

raxhvl commented on docs/wiki/Cryptography/elliptic-curve-cryptography.md:
```diff
@@ -0,0 +1,283 @@
+ 3 + 5 = 8 \in \mathbb{R}
+$$
+
+Elliptic curves are interesting because the points form a field, the result of "addition" of two points remains on the curve. This geometric addition, distinct from arithmetic counterparts, involves drawing a line through chosen points (**P** and **Q**) and reflecting the resulting curve intersection(**R'**) across the x-axis to yield their sum (**R**).
```
Nice catch! Thanks for carefully reviewing. I have:
1. Corrected the highlighted sentence.
2. Added a subsection for group.

raxhvl commented on docs/wiki/Cryptography/elliptic-curve-cryptography.md:
```diff
@@ -0,0 +1,283 @@
+An elliptic curve is a curve **described by the equation**:
+
+$$
+y^2 = x^3 + ax+b
```
Done! Let me know if it looks good.

SiddharthV1 commented on docs/wiki/Cryptography/elliptic-curve-cryptography.md:
```diff
@@ -0,0 +1,297 @@
+## Groups and Fields
+
+### Group
+In mathematics, a **GROUP** is a set $G$, containing at least two elements, which is closed under a binary operation usually referred to as **addition** ($+$). A set is closed under an operation when the result of the operation is also a member of the set. 
```
Shouldn't something like this be also included in group definition?   ... Groups have an associative operation and an identity element

raxhvl replied:
Ideally it should be. My intent is to make as simple as possible for everyone, regardless of their math background. This definition assumes the reader knows about what inverses while the knowledge of inverse and associative is not applied in the context of ECDSA. Does that sound okay?

SiddharthV1 commented on docs/wiki/Cryptography/elliptic-curve-cryptography.md:
```diff
@@ -0,0 +1,297 @@
+## Field
+Similarly, a **FIELD** is a set $F$, containing at least two elements, which is closed under two binary operations usually referred to as **addition** ($+$), and **multiplication**($\times$). 
+
+In other words, A **FIELD** is a **GROUP** under both addition and multiplication.
```
Would it be good to include something like this : A field is an commutative ring containing  multiplicative inverses for non zero elements . Additionally, I think I found a good link giving all three definitions groups , rings and fields on a page : https://www.brainkart.com/article/Groups,-Rings,-and-Fields_8402/

raxhvl replied:
Great resource. The diagram towards the bottom is especially usefull. 

Do you have a reference which cover some examples too maybe like this one: https://www.rareskills.io/post/group-theory-and-coding ?

SiddharthV1 replied:
Yes that's a great idea! let me poke around , if I find something worthwhile I'll post it in this comments thread

raxhvl commented:
hey @taxmeifyoucan I have updated the article:
1. Added a super short intro to public key cryptography.
2. Introduced ECDSA in the context of Ethereum.
3. Added a link to a post that discusses how raw transactions are signed using ECDSA.

taxmeifyoucan commented:
Thanks a lot for the content and updates @raxhvl! It looks good to me, I will add it to the next wiki release
